pkg/api/configsync/v1beta1/resource_override.go

```diff
@@ -92,6 +92,30 @@ type RootSyncOverrideSpec struct {
 	// +kubebuilder:validation:Enum=implicit;explicit
 	// +optional
 	NamespaceStrategy configsync.NamespaceStrategy `json:"namespaceStrategy,omitempty"`
+
+	// clusterRole controls which role to bind the service account for this RootSync's
+	// reconciler to.
+	//
+	// +optional
+	ClusterRole ClusterRoleOverrideSpec `json:"clusterRole,omitempty"`
+}
+
+// ClusterRoleOverrideSpec allows to override settings for the root reconciler permissions
+type ClusterRoleOverrideSpec struct {
+	// name is the name of the ClusterRole to bind to; defaults to "cluster-admin".
```

Comment:
Sorry I got a new round of thoughts after not looking at this over the weekend 😅 

I'm thinking it may be better form to use the following:
```
// +kubebuilder:default:=cluster-admin
```
This would clearly document the default value on the field as well as provide the default behavior at the API level.

---

Additional consideration:

With this realization I'm also considering that we could go back to a single string field and treat it as "disabled" if the user sets the value to `""` (empty string). WDYT?

Reply:
Correction: it looks like the default value only applies if the nested struct (in this case `override`) is set. So that means if the user has an empty override field, the default value would not apply (and `namespaceStrategy` would be `""`). Given this, it looks like we would still need the defaulting logic in the controller and wouldn't be able to rely on an empty string check.